doc/md/guides/postgres/ar-101.md

```diff
@@ -0,0 +1,206 @@
+
+For example, let’s consider a table with four `smallint` and one `bigint` type columns. When a column with a smaller size, such as a `smallint` (2 bytes), has a consecutive column with a larger size, such as `bigint` (8 bytes), Postgres adds 6 bytes of padding to the column of type `smallint` in order to fulfill an alignment requirement of 8 bytes for the consecutive `bigint` column. However, the space used in padding (6 bytes) can be utilized by rearranging the three other `smallint` type columns from the table before the `bigint` type column:
+
+![wrong](https://atlasgo.io/uploads/images/24bytes2.png)
+
+![correct](https://atlasgo.io/uploads/images/16bytes2.png)
+
+:::info
+The catalog pg_type stores information about alignment requirements for various data types. You can access the `typealign` column of this catalog table in order to determine alignment requirements for data types. To know more about the catalog pg_type and interpretation of the  `typealign` column, visit the official documentation [here](https://www.postgresql.org/docs/current/catalog-pg-type.html).
+:::
+
+### Save up on precious space with optimal data alignment
+
+Creating a table with optimal data alignment can help minimize the amount of required disk space. This document describes how we can save space in a database just by changing the order of columns. Let’s see it in action.
+
+### Example
+
+Consider the following table on a 64-bit system.
+
+```sql
+CREATE TABLE users (
+    user_id bigint PRIMARY KEY,
+    active_status boolean,
+    data_balance integer,
+    plan_period smallint                  
+);
+```
+
+Here is what a portion of the table might look like after inserting values:
+
+```sql
+SELECT 
+    * 
+FROM
+    users
+```
+
+```console title="Output"
+ user_id | active_status | data_balance | plan_period 
+---------+---------------+--------------+-------------
+       0 | f             |       731147 |         267
+       1 | t             |       901448 |          26
+       2 | t             |       496823 |          24
+       3 | t             |       985877 |         280
+       4 | f             |       857124 |          83
+.
+.
+.
+ 1999995 | f             |       993259 |          83
+ 1999996 | f             |       926033 |         332
+ 1999997 | f             |       889133 |         280
+ 1999998 | t             |       943793 |           7
+ 1999999 | t             |       550192 |         146
+```
+
+Each tuple in the table takes up 24 bytes of total memory without the header. The `user_id` occupies the first 8 bytes. 
+
+Here, the `active_status` column is of type boolean and takes up only one byte. However, because the next consecutive column `data_balance` is an `integer` with size of 4 bytes, PostgreSQL automatically adds 3 bytes of padding to `active_status` so its size can match with the 4 bytes size of the consecutive `data_balance` column . 
```

Comment:
```suggestion
Here, the `active_status` column is of type boolean and takes up only one byte. However, because the next consecutive column `data_balance` is an `integer` with size of 4 bytes, PostgreSQL automatically adds 3 bytes of padding to `active_status` so its size can match with the 4 bytes size of the consecutive `data_balance` column. 
```